### PR TITLE
JAMES-2935 Add test to ensure that over-specific filter does not exclude valid matches in GetMessagesList

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java [hunk 1]
```diff
@@ -889,6 +898,30 @@ public void getMessageListShouldExcludeMessagesWhenInMailboxesFilterMatches() th
             .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
     }
 
+    @Test
```
Comment (chibenwa) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java]: Can we add it in junit group to run it on top of Cassandra-es?

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java [hunk 2]
```diff
@@ -889,6 +898,30 @@ public void getMessageListShouldExcludeMessagesWhenInMailboxesFilterMatches() th
...
+            .map(ComposedMessageId::getMessageId)
+            .map(MessageId::serialize)
+            .collect(Collectors.toList());
+
+        await();
+
+        given()
+            .header("Authorization", aliceAccessToken.serialize())
+            .body("[[\"getMessageList\", {\"filter\":{\"inMailboxes\":[\"" + mailboxId.serialize() + "\"],\"text\":\"test\"}}, \"#0\"]]")
+            .when()
```
Comment (chibenwa) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java]: Indent

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java (hunk 1 above)
Comment (blackheaven) [reply server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/GetMessageListMethodTest.java]: `@Category(BasicFeature.class)`?

**Comment by Arsnael** [comment]
```
[ERROR]   SmtpRandomStoringTest.oneHundredMailsShouldHaveBeenStoredBetweenFourAndEightTimes:168 » ConditionTimeout
```
I noticed this one failing couple of times this last few days, we should probably take a look...

test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**